Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -387,6 +387,16 @@ public AndConstraint<TAssertions> NotMatch(string wildcardPattern, string becaus
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        public AndConstraint<TAssertions> MatchRegex([RegexPattern][StringSyntax("Regex")] string regularExpression,
+            OccurrenceConstraint occurrenceConstraint, string because = "", params object[] becauseArgs)
+        {
+            MatchRegex(regularExpression, because, becauseArgs);
```

Comment:
We should not call the regular `MatchRegex` overload first as it would make this test fail.
`"a".Should().MatchRegex("b", Times.Exactly(0))`

Reply:
ok, I see.. working on a fix in approximately 3 hours..